daemon/network.go

```diff
@@ -1005,49 +1005,43 @@ func getPortMapInfo(sb *libnetwork.Sandbox) nat.PortMap {
 	}
 
 	for _, ep := range sb.Endpoints() {
-		pm, _ = getEndpointPortMapInfo(ep)
-		if len(pm) > 0 {
-			break
-		}
+		getEndpointPortMapInfo(pm, ep)
 	}
 	return pm
 }
 
-func getEndpointPortMapInfo(ep *libnetwork.Endpoint) (nat.PortMap, error) {
-	pm := nat.PortMap{}
-	driverInfo, err := ep.DriverInfo()
-	if err != nil {
-		return pm, err
-	}
-
+func getEndpointPortMapInfo(pm nat.PortMap, ep *libnetwork.Endpoint) {
+	driverInfo, _ := ep.DriverInfo()
 	if driverInfo == nil {
 		// It is not an error for epInfo to be nil
-		return pm, nil
+		return
 	}
 
-	if expData, ok := driverInfo[netlabel.ExposedPorts]; ok {
-		if exposedPorts, ok := expData.([]lntypes.TransportPort); ok {
-			for _, tp := range exposedPorts {
-				natPort, err := nat.NewPort(tp.Proto.String(), strconv.Itoa(int(tp.Port)))
-				if err != nil {
-					return pm, fmt.Errorf("Error parsing Port value(%v):%v", tp.Port, err)
+	if containertypes.NetworkMode(ep.Network()).IsBridge() {
```

Comment:
To be honest, I feel like this is a bit crappy as this is going to cause issues when we'll add support for DNS-based links to custom networks. But this is the shortest path to get this bug fixed in v28.0.2.

Reply:
Maybe… I think one part of the problem here is that we're using a single variable to report two distinct concepts (ie. 'exposed ports' and 'published ports').

Currently you can't connect a container to the default bridge and to a custom network, so a container can't have port-mappings provided by two networks and legacy links at the same time.

But once we have DNS-based links available, this will become possible and we'll realize that 'exposed ports' should be an endpoint property. So, maybe it's not worth adding a TODO for a block of code which will be deleted without getting fixed?

Reply:
Done